lib/loaders/config-file.js

```diff
@@ -57,5 +58,59 @@ module.exports = function(filename) {
 
   // Passing an object as the config value merges
   // the config data
-  return _.assign(config, configObject);
+  var compiled = _.assign(config, configObject);
+
+  // Check the rewrites/redirects/headers for fields that might break based on
+  // Firebase Hosting accepting only RE2, but RE2 being an optional dependency
+  // in Superstatic.
+  if (_.isArray(compiled.rewrites)) {
```

Comment:
I don't think I actually want `superstatic` to print these messages since it limits our ability to manage them directly from the Firebase CLI or other callers that use this as a library.

Is there a simple way to expose that we're in "RE2 emulation mode" to a caller instead? I don't know that we need such detailed error messages so much as we need to maybe have a section in the README that documents the behavior when RE2 is/isn't available, for `superstatic` to be silent but make it possible to detect when that's the case, and for the CLI to use that detection method to warn people / point them to the README.

Reply:
Yes, quite easily. Something like `exports.RE2mode = require('./utils/patterns').re2Available;` in index.js would do the job nicely.